experimental/LieAlgebras/src/root_systems.jl

```diff
@@ -0,0 +1,164 @@
+mutable struct RootSystem <: AbstractVector{fmpq}
+  roots::Vector{Vector{Int}}
+  simple_roots::Vector{Vector{Int}}
+  positive_roots::Vector{Vector{Int}}
+  root_system_type::Vector{Union{String, Int64}}
+  @doc raw"""
+    `RootSystem(S::String) -> RootSystem`
+  Return the root system of type `S` where `S` is a string consisting out of
+  a letter `A`, `B`, `C`, `D`, `E`, `F`, `G` followed by an integer.
+  For the exceptional root system, the integers are fixed, e.g. `G2`, `F4`, `E6`, ...
+  """
+  function RootSystem(S::String)
+    # S is a string detailing the type of the indecomposable root system made out of a letter, e.g. "A", "B", "C",... and an integer for the number of simple roots
+    l = length(S)
+    n = parse(Int64, S[2:l])
+    S1 = GAP.Obj(string(S[1]))
+    RS = GAP.Globals.RootSystem(S1, n)
+    Ro_1 = GAP.Globals.PositiveRoots(RS)
+    Ro_2 = GAP.Globals.NegativeRoots(RS)
+    sR = GAP.Globals.SimpleSystem(RS)
+    sR = [[sR[i][j] for j=1:length(sR[i])] for i=1:length(sR)]
+    Ro1 = [[Ro_1[i][j] for j=1:length(Ro_1[i])] for i=1:length(Ro_1)]
+    Ro2 = [[Ro_2[i][j] for j=1:length(Ro_2[i])] for i=1:length(Ro_2)]
+    S = Union{String, Int64}[S[1:1], n]
+    Ro = reduce(vcat, (Ro1, Ro2))
+    return new(Ro,sR,Ro1,S)
+  end
+end
+
+###############################################################################
+#
+#   Basic manipulation
+#
+###############################################################################
+ @doc raw"""
+  `size(R::RootSystem, [dim::Int])`
+ Return a tuple containing the dimensions of the vector of roots, i.e. the number of roots in the first entry. 
+ Optionally you can specify a dimension to just get the length of that dimension.
+  """
```

Comment:
```suggestion
@doc raw"""
    size(R::RootSystem, [dim::Int])

Return a tuple containing the dimensions of the vector of roots, i.e. the number of roots in the first entry. 
Optionally you can specify a dimension to just get the length of that dimension.
"""
```

Reply:
Note that `size` already has a very different meaning in Julia; I am not convinced it is a good idea to re-use this name here. 

How about `number_of_roots` ?

Regarding the option `dim`, I don't understand what this about. What is "the length of that dimension" ?!

Also, instead of "root in the first entry", how about "roots in the root system `R`" ?

Reply:
I have changed the RootSystem type, so that it is no longer a subtype of AbstractVector and doesn't need the size function anymore.